src/loaders/stm_load.c

```diff
@@ -270,7 +270,8 @@ static int stm_load(struct module_data *m, HIO_HANDLE * f, const int start)
 	D_(D_INFO "Stored samples: %d", mod->smp);
 
 	for (i = 0; i < mod->ins; i++) {
-		if (mod->xxs[i].len > 1) {
+		if (sfh.ins[i].volume && sfh.ins[i].length) {
```

Comment:
Yes, I'm not arguing against seeking to the correct sample position based on "rsvd1" -- in fact, I have done the same in libopenmpt/OpenMPT for a long time already. I'm just saying that whether you load samples with a default volume of 0 or not doesn't really matter because when you try to play such a sample in ST2, it results in undefined behaviour (noise is played).